src/Surface.zig

```diff
@@ -3383,6 +3389,12 @@ pub fn cursorPosCallback(
         self.renderer_state.terminal.screen.dirty.hyperlink_hover = true;
     }
 
+    // Stop selection scrolling when inside the viewport within a 1px buffer
+    // for fullscreen windows, but only when selection scrolling is active.
+    if (pos.x >= 1 and pos.y >= 1 and self.io_thread.scroll_active) {
```

Comment:
Accessing the IO thread state here is a data race and is unsafe. I think we need to store this state on the Surface itself.